web/client/src/library/components/plan/PlanApplyStageTracker.tsx

```diff
@@ -0,0 +1,819 @@
+            Given a plan, it pushes snapshots into the state and then kicks off
+            the backfill process for all affected snapshots. Once backfill is
+            done, snapshots that are part of the plan are promoted in the
+            environment targeted by this plan.
+          </small>
+        </>
+      )}
+      <div className="py-2">{children}</div>
+      {isNotNil(end) && (
+        <>
+          <Divider />
+          <small className="text-neutral-500 block px-4 mt-1">
+            Evaluation stopped at{' '}
+            <b>{toDateFormat(new Date(end), 'yyyy-mm-dd hh-mm-ss')}</b>
+          </small>
+        </>
+      )}
+    </div>
+  )
+}
+
+function StageCreation({
+  report,
+  isOpen,
+}: {
+  report: PlanStageCreation
+  isOpen?: boolean
+}): JSX.Element {
+  return (
+    <Stage
+      meta={report.meta!}
+      states={[
+        'Snapshot Tables Created',
+        'Snapshot Tables Creation Failed',
+        'Creating Snapshot Tables...',
+      ]}
+      isOpen={isOpen}
+    >
+      {isNotNil(report.total_tasks) && (
+        <TasksOverview.Block>
+          <TasksOverview.Task>
+            <TasksOverview.TaskDetails>
+              <TasksOverview.TaskInfo>
+                <TasksOverview.TaskHeadline headline="Snapshot Tables" />
+              </TasksOverview.TaskInfo>
+              <TasksOverview.DetailsProgress>
+                <TasksOverview.TaskSize
+                  completed={report.total_tasks}
+                  total={report.num_tasks}
+                  unit="task"
+                />
+                <TasksOverview.TaskDivider />
+                <TasksOverview.TaskProgress
+                  completed={report.num_tasks}
+                  total={report.total_tasks}
+                />
+              </TasksOverview.DetailsProgress>
+            </TasksOverview.TaskDetails>
+            <Progress
+              progress={toRatio(report.num_tasks, report.total_tasks)}
+            />
+          </TasksOverview.Task>
+        </TasksOverview.Block>
+      )}
+    </Stage>
+  )
+}
+
+function StageRestate({ report }: { report: PlanStageRestate }): JSX.Element {
+  return (
+    <Stage
+      meta={report.meta!}
+      states={[
+        'Restate Models',
+        'Restate Models Failed',
+        'Restating Models...',
+      ]}
+    />
+  )
+}
+
+function StageBackfill({
+  environment,
+  backfill,
+  isVirtualUpdate,
+  backfills,
+}: {
+  environment: string
+  backfill: PlanStageBackfill
+  isVirtualUpdate: boolean
+  backfills?: PlanStageBackfills
+}): JSX.Element {
+  const { change_categorization } = usePlan()
+
+  const categories = useMemo(
+    () =>
+      Array.from(change_categorization.values()).reduce<
+        Record<string, boolean[]>
+      >((acc, { category, change }) => {
+        change?.indirect?.forEach(model => {
+          if (acc[model] == null) {
+            acc[model] = []
+          }
+
+          acc[model]?.push(category.value !== 1)
```

Comment:
What does the 1 (and 3 below) mean?

Reply:
so openapi SnapshotChangeCategory converts to 
```
export const SnapshotChangeCategory = {
  NUMBER_1: 1, // BREAKING
  NUMBER_2: 2, // NON_BREAKING
  NUMBER_3: 3, // FORWARD_ONLY
  NUMBER_4: 4, // INDIRECT_BREAKING
  NUMBER_5: 5, // INDIRECT_NON_BREAKING
} as const
```